client/src/clients/decorators/proposervmApi.ts

```diff
@@ -0,0 +1,65 @@
+import { Chain, Transport } from "viem";
+import { getCurrentEpoch } from "../../methods/proposervm/getCurrentEpoch.js";
+import { getProposedHeight } from "../../methods/proposervm/getProposedHeight.js";
+import { GetCurrentEpochReturnType } from "../../methods/proposervm/types/getCurrentEpoch.js";
+import { GetProposedHeightReturnType } from "../../methods/proposervm/types/getProposedHeight.js";
+import { AvalancheCoreClient } from "../createAvalancheCoreClient.js";
+
+export type ProposervmAPIActions = {
+  /**
+   * Returns this node's current proposer VM height.
```

Comment:
nit: This API will return node's proposervm height specific to the queried chain (X/P/C/L1s).